graql/reasoner/atom/task/validate/RelationAtomValidator.java

```diff
@@ -65,9 +70,10 @@ public void checkValid(RelationAtom atom, ReasoningContext ctx) {
     @Override
     public Set<String> validateAsRuleHead(RelationAtom atom, Rule rule, ReasoningContext ctx) {
         //can form a rule head if type is specified, type is not implicit and all relation players are insertable
-        return Sets.union(
+        Set<String> errors = Sets.union(
```

Comment:
ah so that's the bit when you want to have union with 3 sets. It might be clearer to just initialise a set to empty and then do `.addAll` one by one. You would immediately see that you have a union of 3 things.